banzai/main.py

```diff
@@ -241,126 +238,55 @@ def reduce_night():
+    elif image_suffix == 'f00.fits':
+        stages = get_stages_todo(last_stage=dark.DarkSubtractor,
+                                 extra_stages=[flats.FlatNormalizer, qc.PatternNoiseDetector, flats.FlatComparer])
+    else:
+        stages = get_stages_todo()
+    return stages
 
 
 def run_preview_pipeline():
     parser = argparse.ArgumentParser(
         description='Make master calibration frames from LCOGT imaging data.')
-
-    parser.add_argument("--processed-path", default='/archive/engineering',
-                        help='Top level directory where the processed data will be stored')
-    parser.add_argument("--log-level", default='debug', choices=['debug', 'info', 'warning',
-                                                                 'critical', 'fatal', 'error'])
-    parser.add_argument('--post-to-archive', dest='post_to_archive', action='store_true',
-                        default=False)
-    parser.add_argument('--post-to-elasticsearch', dest='post_to_elasticsearch', action='store_true',
-                        default=False)
-    parser.add_argument('--db-address', dest='db_address',
-                        default='mysql://cmccully:password@localhost/test',
-                        help='Database address: Should be in SQLAlchemy form')
-    parser.add_argument('--fpack', dest='fpack', action='store_true', default=False,
-                        help='Fpack the output files?')
-    parser.add_argument('--rlevel', dest='rlevel', default=11, help='Reduction level')
     parser.add_argument('--n-processes', dest='n_processes', default=12,
                         help='Number of listener processes to spawn.', type=int)
     parser.add_argument('--broker-url', dest='broker_url',
                         default='amqp://guest:guest@rabbitmq.lco.gtn:5672/',
                         help='URL for the broker service.')
     parser.add_argument('--queue-name', dest='queue_name', default='preview_pipeline',
                         help='Name of the queue to listen to from the fits exchange.')
-    parser.add_argument('--max-preview-tries', dest='max_preview_tries', default=5,
-                        help='Maximum number of tries to produce a preview image.')
-    parser.add_argument('--elasticsearch-url', dest='elasticsearch_url',
-                        default='http://elasticsearch.lco.gtn:9200')
-    parser.add_argument('--es-index', dest='elasticsearch_qc_index', default='banzai_qc',
-                        help='ElasticSearch index to use for QC results')
-    parser.add_argument('--es-doc-type', dest='elasticsearch_doc_type', default='qc',
-                        help='Elasticsearch document type for QC records')
-    parser.add_argument('--no-bpm', dest='no_bpm', default=False, action='store_true',
-                        help='Do not use a bad pixel mask to reduce data (BPM contains all zeros)')
 
-    args = parser.parse_args()
-    args.preview_mode = True
```

Comment:
Shouldn't preview mode be set somewhere?

Reply:
This is indeed now set.